cxx11.cmake

```diff
@@ -0,0 +1,47 @@
+# Copyright (C) 2018 LAAS-CNRS, JRL AIST-CNRS, INRIA
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+include(CheckCXXCompilerFlag)
+
+#.rst:
+# .. ifmode:: user
+#
+# .. command:: PROJECT_USE_CXX11
+#
+#    This macro set up the project to compile the whole project 
+#    with C++11 standards.  
+#
+MACRO(PROJECT_USE_CXX11)
+  CHECK_CXX_COMPILER_FLAG("-std=c++0x" COMPILER_SUPPORTS_CXX0X)
+  CHECK_CXX_COMPILER_FLAG("-std=c++11" COMPILER_SUPPORTS_CXX11)
+  IF(COMPILER_SUPPORTS_CXX0X OR COMPILER_SUPPORTS_CXX11)
+    IF(CMAKE_VERSION VERSION_LESS "3.1")
+      IF(COMPILER_SUPPORTS_CXX0X)
+        SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++0x")
+      ELSE()  
```

Comment:
Should check `COMPILER_SUPPORT_CXX11` here and simply don't add a flag if the compiler supports neither (that's the case for MSVC)

Reply:
This is what it is done but not explicitly for cxx11

Reply:
Sorry, I missed the initial condition check.

I would prefer this to be done only for CMake < 3.1, i.e. instead of the current:
```
CHECK_FLAGS()
IF(SUPPORT_FLAGS)
  IF(CMAKE_VERSION < 3.1)
    ...
  ELSE()
    ...
  ENDIF()
ENDIF()
```

I'd rather we do:
```
IF(CMAKE_VERSION < 3.1)
  CHECK_FLAGS()
  IF(SUPPORT)
    ..
  ELSE()
    // Maybe raise a cmake error here?
  ENDIF()
ELSE()
  ...
ENDIF()
```

Reply:
But we have to check the flags anyway, because `CMAKE_CXX_STANDARD` is not checking that for us.
I think it is better to provide a feedback to users at this stage.

Reply:
We could use `CXX_STANDARD_REQUIRED` in that case (see https://cmake.org/cmake/help/v3.1/prop_tgt/CXX_STANDARD_REQUIRED.html)

The whole problem is that `-std=c++0x` or `-std=c++11` is not supported by all compilers (most notably/importantly visual studio) although they might support C++11 or better. Also, checking compiler flags on Visual Studio/Windows is lengthy.

Reply:
It has been added to 59bcf58.

Reply:
As I'm not using Windows, I'm not able to adjust the right strategy.
I would prefer to fix in a future PR.